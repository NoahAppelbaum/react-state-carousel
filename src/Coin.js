const HEADS_IMG = "https://upload.wikimedia.org/wikipedia/commons/thumb/3/33/2010_cent_obverse.jpg/240px-2010_cent_obverse.jpg"
const TAILS_IMG = "https://upload.wikimedia.org/wikipedia/commons/thumb/7/72/Jefferson-Nickel-Unc-Obv.jpg/244px-Jefferson-Nickel-Unc-Obv.jpg"

/** Coin: Display heads or tails
 *
 * Props -
 * isHeads: boolean (true/false)
 *
 * Flipper --> Coin
 */
function Coin({ isHeads }) {
  const source = isHeads ? HEADS_IMG : TAILS_IMG;
  const altText = isHeads ? "heads" : "tails";

  return (
    <div className="Coin">
      <img src={source} alt={altText} />
    </div>
  )

}

export default Coin