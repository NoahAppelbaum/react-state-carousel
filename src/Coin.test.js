import Coin from "./Coin";
import { render } from "@testing-library/react";

it("renders without crashing", function () {
  render(<Coin />);
});

it("should match snapshot", function () {
  const { container, debug } =
    render(<Coin
      isHeads={true} />);

  expect(container).toMatchSnapshot();
});
