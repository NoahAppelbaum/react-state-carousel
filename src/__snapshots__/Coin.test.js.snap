// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="Coin"
  >
    <img
      alt="heads"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/33/2010_cent_obverse.jpg/240px-2010_cent_obverse.jpg"
    />
  </div>
</div>
`;
